pkg/cloud/services/compute/instance.go

```diff
@@ -233,6 +235,16 @@ func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infr
 		return nil, fmt.Errorf("no ports with fixed IPs found on Subnet \"%s\"", i.Subnet)
 	}
 
+	for _, portCreateOpts := range i.Ports {
+		port, err := getOrCreatePort(is, i.Name+"-"+portCreateOpts.NameSuffix, portCreateOpts)
+		if err != nil {
+			return nil, err
```

Comment:
should we delete those ports when delete cluster? I assume those ports are independent to VM lifecylce?

Reply:
I implemented a cleanup orphaned port logic in one of my PRs before the sec group deletion. But it would be good if we try to delete all ports during machine deletion which we created for this machine before.